### PR TITLE
ci: add NMP customizer contract test configs

### PR DESCRIPTION
**Pull request by adil-a**
## Summary
- Add 13 NMP customizer contract test configs for nightly CI
- These serve as **contract tests** — if any stop working with `finetune.py`, it means a breaking change was introduced that affects the customizer integration
- Configs organized into existing model-family directories with `customizer_` filename prefix
- Integrated into existing model sections in `nightly_recipes.yml`
- Launcher detects customizer configs via `*customizer_*` filename glob to inject CI dataset overrides

## CI Infrastructure Changes
- `finetune_launcher.sh`: per-config `nproc_per_node` override via `CONFIG_NPROC_PER_NODE` env var, allowing single-GPU and 4-GPU configs to run on 8-GPU CI nodes
- `generate_ci_tests.py`: map `ci.nproc_per_node` YAML key to `CONFIG_NPROC_PER_NODE` job variable
- `README.md`: document `nproc_per_node` CI option

## Bug Fixes (discovered during validation)
- **Robustness test lr_scheduler mismatch**: Phase 6 resume test baseline computed a different `lr_decay_steps` than Phase 1 (due to different `max_steps`), causing divergent LR curves and false loss mismatches. Fixed by pinning baseline `lr_decay_steps` to match Phase 1.
- **Chat template OSError**: `_resolve_chat_template()` called `Path(template).exists()` on long inline Jinja2 strings, exceeding Linux NAME_MAX. Fixed with early return for Jinja2 markers.
- **Nemotron PEFT out_proj**: Mamba mixer `out_proj` layers use custom CUDA kernels incompatible with LoRA. Added `exclude_modules: ["*.out_proj"]` matching existing nemotron PEFT configs.

## Config Fixes
- Nemotron PEFT: `dp_size=null, ep_size=4` for proper expert parallelism (was `dp_size=1, ep_size=null`)
- Removed `lr_scheduler` from packing configs (small packed dataset with `num_epochs=1` causes `lr_decay_steps=0`)
- Bumped `num_epochs` for small-dataset configs so robustness test baseline has enough steps
- Loosened `hf_kl_threshold` to `1e-2` for full-SFT configs (bf16 rounding in consolidated checkpoint vs HF loading)

## Validation
All 10 non-chat configs validated via `finetune_launcher.sh` robustness commands on cw-dfw (8x H100 80GB). Chat configs (3) not yet validated — CI sample chat dataset is missing `tool_calls[].id` field required by the tokenizer; tracked separately.

## Test Matrix

| Model | SFT | SFT+Chat | PEFT | PEFT+Packing |
|---|---|---|---|---|
| GPT-OSS 20B (MoE, 8 GPU) | :white_check_mark: | :x: not validated | :white_check_mark: (1 GPU) | :white_check_mark: (1 GPU) |
| Llama 3.1 8B (TP=2, 8 GPU) | :white_check_mark: | | | |
| Llama 3.2 1B (1 GPU) | :white_check_mark: | :x: not validated | :white_check_mark: | :white_check_mark: |
| Nemotron Nano V3 (MoE, 4/8 GPU) | :white_check_mark: | :x: not validated | :white_check_mark: | :white_check_mark: |

## Notes
- Configs sourced from the NMP customizer service compile_automodel_config() output with ci: sections added
- Chat configs include inline Jinja2 chat_template (model-specific, generated by the customizer)
- Sample datasets placed on CI cluster; data paths overridden via CLI args at runtime

### CODE REVIEW

File: examples/llm_finetune/gpt_oss/customizer_gpt_oss_full_sft.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+dist_env:
```
Comment (thomasdhc) [inline examples/llm_finetune/gpt_oss/customizer_gpt_oss_full_sft.yaml:3]: @adil-a Can we organize the customizer configs into examples/customizer/_model_ pathway. This will make it easier for integrating with the existing cicd setup

File: tests/ci_tests/configs/llm_finetune/nightly_recipes.yml [hunk 2]
```diff
@@ -39,6 +39,8 @@ configs:
   # -- gpt_oss --
   - gpt_oss/gpt_oss_20b.yaml
   - gpt_oss/gpt_oss_20b_single_gpu.yaml
+  - gpt_oss/customizer_gpt_oss_full_sft.yaml
```
Comment (thomasdhc) [inline tests/ci_tests/configs/llm_finetune/nightly_recipes.yml]: Let's moves these out of nightly, should be automatically picked up by release test suite. Our nightly test suite is getting too large.

**Review by akoumpa — APPROVED** [review]
